examples/moderation_example.py

```diff
@@ -21,51 +21,69 @@
 
 import lightbulb
 
-bot = lightbulb.BotApp(token="YOUR_TOKEN", intents=hikari.Intents.ALL_UNPRIVILEGED)
+bot = hikari.GatewayBot(token="...")
+client = lightbulb.client_from_app(bot)
 
+bot.subscribe(hikari.StartingEvent, client.start)
 
-@bot.command()
-@lightbulb.option("reason", "Reason for the ban", required=False)
-@lightbulb.option("user", "The user to ban.", type=hikari.User)
-@lightbulb.command("ban", "Ban a user from the server.")
-@lightbulb.implements(lightbulb.SlashCommand)
-async def ban(ctx: lightbulb.SlashContext) -> None:
-    """Ban a user from the server with an optional reason."""
-    if not ctx.guild_id:
-        await ctx.respond("This command can only be used in a guild.")
-        return
 
-    # Create a deferred response as the ban may take longer than 3 seconds
-    await ctx.respond(hikari.ResponseType.DEFERRED_MESSAGE_CREATE)
-    # Perform the ban
-    await ctx.app.rest.ban_user(ctx.guild_id, ctx.options.user.id, reason=ctx.options.reason or hikari.UNDEFINED)
-    # Provide feedback to the moderator
-    await ctx.respond(f"Banned {ctx.options.user.mention}.\n**Reason:** {ctx.options.reason or 'No reason provided.'}")
+@client.register()
+class Ban(
```

Comment:
It would be nice if you could showcase app permissions by passing `default_member_permissions=hikari.Permissions.BAN_MEMBERS` and `dm_enabled=False` for this command, as well as the appropriate permission for the purge command (`MANAGE_MESSAGES`)